lib/sentry/transport.ex

```diff
@@ -86,6 +87,21 @@ defmodule Sentry.Transport do
     kind, data -> {:error, {kind, data, __STACKTRACE__}}
   end
 
+  defp client_post_and_validate_return_value(client, endpoint, headers, body) do
+    case client.post(endpoint, headers, body) do
+      {:ok, status, resp_headers, resp_body}
+      when is_integer(status) and status in 200..599 and is_list(resp_headers) and
+             is_binary(resp_body) ->
+        {:ok, status, resp_headers, resp_body}
+
+      {:ok, _status, _resp_headers, _resp_body} ->
+        {:error, :invalid_http_response}
```

Comment:
I don't think this is a fair error code: we don't know that the HTTP response is invalid. What's invalid here is that the HTTP client didn't comply to the expected behaviour. What I would do here is return

```elixir
{:error, {:malformed_http_client_response, status, resp_headers, resp_body}}`
```

so that it makes it sort of understandable for people?

Reply:
Yes, much better 💟